modules/core/src/lib/widget-manager.ts

```diff
@@ -57,61 +63,119 @@ const ROOT_CONTAINER_ID = '__root';
 export class WidgetManager {
   deck: Deck;
   parentElement?: HTMLElement | null;
-  containers: {[id: string]: HTMLDivElement} = {};
-  widgets: Widget[] = [];
-  lastViewports: {[id: string]: Viewport} = {};
+
+  /** Widgets added via the imperative API */
+  private defaultWidgets: Widget[] = [];
```

Comment:
Maybe the names could be more reflective of the comments - `imperativeWidgets` / `declarativeWidgets` / `allWidgets` etc?

Reply:
Yes maybe.. there's ongoing discussion in a new RFC on what our imperative API could be.